cli/waiter/data_format.py

```diff
@@ -0,0 +1,148 @@
+import json
+import logging
+import os
+import sys
+
+import yaml
+
+
+class DataFormat:
+    def __str__(self):
+        """Returns the name of the format in upper case."""
+        return self.name().upper()
+
+    def name(self):
+        """Returns the name of the format."""
+        raise NotImplementedError('Method has not been implemented!')
+
+    def parse(self, data):
+        """Returns the parsed data."""
+        raise NotImplementedError('Method has not been implemented!')
+
+    def dump(self, out_data, out_file=None):
+        """
+        If the file is provided, writes the string representation of the data to the file.
+        Else it returns the string representation of the data in the format.
+        """
+        raise NotImplementedError('Method has not been implemented!')
+
+
+class JsonDataFormat(DataFormat):
+    def name(self):
+        return 'json'
+
+    def parse(self, data):
+        try:
+            logging.debug(f'parsing input data as json')
+            content = json.loads(data)
+            return content
+        except Exception:
+            raise ValueError('Malformed JSON in input.')
+
+    def dump(self, out_data, out_file=None):
+        if out_file:
+            json.dump(out_data, out_file, indent=2, sort_keys=True)
+        else:
+            return json.dumps(out_data, indent=2, sort_keys=True)
+
+
+class YamlDataFormat(DataFormat):
+    def name(self):
+        return 'yaml'
+
+    def parse(self, data):
+        try:
+            logging.debug(f'parsing input data as yaml')
+            content = yaml.safe_load(data)
+            return content
+        except Exception:
+            raise ValueError('Malformed YAML in input.')
+
+    def dump(self, out_data, out_file=None):
+        if out_file:
+            yaml.safe_dump(out_data, out_file)
+        else:
+            return yaml.safe_dump(out_data)
+
+
+JSON = JsonDataFormat()
+YAML = YamlDataFormat()
+
+
+def validate_options(options):
+    """Validates whether unique file format is specified."""
+    as_json = options.get(JSON.name())
+    as_yaml = options.get(YAML.name())
+    if as_json and as_yaml:
+        raise Exception(f'JSON and YAML mode cannot be used simultaneously!')
```

Comment:
Would it be possible to use this with `argparse`'s mutual exclusion?

https://docs.python.org/3/library/argparse.html#mutual-exclusion